File: solutions/uri/1022/1022.c
#include <math.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

int main() {
    int16_t n, a, b, c, d, r, num, den, num_r, den_r, j;
    char o, i;

    scanf("%d", &n);
    while (n--) {
        scanf("%d %c %d %c %d %c %d", &a, &i, &b, &o, &c, &i, &d);
        switch (o) {
            case '+':
                num = (b * d) * a / b + (b * d) * c / d;
                den = b * d;
                break;
            case '-':
                num = (b * d) * a / b - (b * d) * c / d;
                den = b * d;
                break;
            case '*':
                num = a * c;
                den = b * d;
                break;
            default:
                num = a * d;
                den = b * c;
                break;
        }

        if (abs(num) < abs(den)) {
            r = abs(num);
        } else {
            r = abs(den);
        }

        num_r = num;
        den_r = den;

        j = 2;
        while (j <= r / 2 + 1) {
            if (num_r % j == 0 && den_r % j == 0) {
                num_r = num_r / j;
                den_r = den_r / j;
            } else {
                j++;
            }
        }

        printf("%d/%d = %d/%d\n", num, den, num_r, den_r);
    }
    return 0;
}

File: solutions/uri/1022/1022.py
import sys

n = int(sys.stdin.readline())

for _ in range(n):
    a, _, b, o, c, _, d = sys.stdin.readline().split()
    a, b, c, d = map(int, [a, b, c, d])

    if o == "+":
        num = (b * d) * a // b + (b * d) * c // d
        den = b * d
    elif o == "-":
        num = (b * d) * a // b - (b * d) * c // d
        den = b * d
    elif o == "*":
        num = a * c
        den = b * d
    else:
        num = a * d
        den = b * c

    r = abs(den)
    if abs(num) < abs(den):
        r = abs(num)

    num_r = num
    den_r = den

    j = 2
    while j <= r // 2 + 1:
        if num_r % j == 0 and den_r % j == 0:
            num_r = num_r // j
            den_r = den_r // j
        else:
            j += 1

    print(f"{num}/{den} = {num_r}/{den_r}")
